### ISSUE TITLE
Port to NetCore (for xplat)

### ISSUE THREAD

**Issue by admalledd** [Rybadour/Foreman#20]
I am not 100% sure, since I am just quickly reading source code and not digging into exact bits as much as I would want before filing this, (may edit more later after feedback) but it seems you aren't intrinsically using anything that can't work on netcore3+ besides WinForms for UI so that Linux (and maybe other?) users can also use the application.

I might take a stab at a number of the first few steps sometime in the next few weeks via PRs, for me the killer will be UI work. Never been very good at that since I work systems stuff.

# Things to do

- [ ] Convert from legacy csproj to SDK style
  - [ ] Include moving to using nuget `<PackageReference>` format instead of `packages.config`
- [ ] Use nuget packages instead of direct DLLs for NLua and friends.
  - [ ] Importantly is for the transient dependency KeriLua which should ref the native DLL/SO files correctly for us on `dotnet publish`. Requires validation to be sure since runtime selection can be picky, so might have to document the correct CLI/VS magic to properly publish-with-native-deps. 
- [ ] Make sure the WinForms stuff still works after convert to NetCore3+ (But is going to go away/reworked with Linux version of something else UI)
- [ ] At this point Forman should work on windows via NetCoreApp3+ publish to exe (but not on Linux/others yet)
- [ ] Investigate cross platform windowing libs, hopefully one is low effort to port from WinForms...
  - [ ] [AvaloniaUI](https://github.com/AvaloniaUI/Avalonia) (kinda but not WPF? )
  - [ ] [Xamarin.Forms](https://github.com/xamarin/Xamarin.Forms) (Sounds like a partial API port of WinForms? or am I very wrong on that?)
  - [ ] [GTK#](https://github.com/GtkSharp/GtkSharp) (Might be better one long-term?)
  - [ ] Is there something *very* cross platform that we could even target web/WASM? Does anything dotnet support full-desktop and web in same API? (Note, Blazor doesn't) This would require `emscripten/C/C++` compiling the native Lua (+Others?) libraries. Or could we swap out NLua for MoonSharp? how compatible do we require? hrm. Very low priority this, because probably better for different approach when targeting web.

**Comment by Rybadour** [comment]
Awesome! Sounds like the main reason for this conversion is to support Linux users?

What is Xplat? A totally different UI layer?

**Comment by admalledd** [comment]
Yes, this is primarily because I want to try this, and I am on Linux :D

Xplat --> x-platform --> cross-platform. I used some terms that I am far more familiar with a bit since I was quickly writing this up while waiting for my work IDE debugger to hit a deep breakpoint. Sorry about that. Generally used as a term when there is more than one, although since this would be basically "just windows/linux" at the end of this, maybe xplat doesn't apply as nicely for this. I'll edit a bit.
